File: python/nt_solutions/coding_1_approximation/exo10.py
plt.figure(figsize = (8,8))

for u in range(len(Mlist)):
    M = Mlist[u]
    a = np.sort(np.ravel(abs(fL)))[::-1] #sort a 1D copy of F in descending order
    T = a[M]
    fLT = np.multiply(fL,abs(fL) > T)
    #fLT = perform_thresholding(fL, M, 'largest')
    fM = fLT
    
    for i in range(1,n//w+1):
        for j in range(1,n//w+1):
            fM[(i-1)*w: i*w, (j-1)*w: j*w] = idct2(fLT[(i-1)*w: i*w, (j-1)*w: j*w])
    
    imageplot(clamp(fM), "M/N = %.2f, SNR = %.1f dB" %(M/n**2,snr(f,fM)),[1, 2, u+1])

File: r/nt_solutions/coding_1_approximation/exo10.R




for (u in 1:length(Mlist)){
  M <- Mlist[u]
  a <- as.vector(abs(fL))[order(-as.vector(abs(fL)))] #sort a 1D copy of F in descending order
  T <- a[M]
  fLT <- fL * (abs(fL) > T)
  #fLT = perform_thresholding(fL, M, 'largest')
  fM <- fLT
  
  for (i in 1:(n%/%w)){
    for (j in 1:(n%/%w)){
      fM[((i-1)*w+1):(i*w), ((j-1)*w+1):(j*w)] <- idct2(fLT[((i-1)*w+1):(i*w), ((j-1)*w+1):(j*w)])
    }
  }
  
  imageplot(clamp(fM), paste("M/N =", format(M/n**2, digits=3),", SNR =", format(snr(f,fM), digits=3),"dB"), c(1, 2, u))
}